fe/fe-core/src/main/java/com/starrocks/catalog/ResourceMgr.java

```diff
@@ -105,6 +105,45 @@ private void onDropResource(Resource resource) {
         }
     }
 
+    /**
+     * alter resource statement only support external hive now .
+     * @param stmt
+     * @throws DdlException
+     */
+    public void alterResource(AlterResourceStmt stmt) throws DdlException {
+        String name = stmt.getResourceName();
+
+        //check if the target resource exists .
+        Resource resource = nameToResource.get(name);
+        if (resource == null) {
+            throw new DdlException("Resource(" + name + ") does not exist");
+        }
+
+        if (resource instanceof HiveResource) {
+            // update the properties
+            Map<String, String> properties = ((HiveResource) resource).getProperties();
+            for (Map.Entry<String, String> entry : stmt.getProperties().entrySet()) {
+                String key = entry.getKey();
+                if (!properties.containsKey(key)) {
+                    throw new DdlException("Property(" + key + ") does not support");
+                }
+                properties.put(key, entry.getValue());
+            }
```

Comment:
1. Better to do this in HiveResource.
2. Why adding new properties is not supported?

Reply:
The user may misspelling the property  that needs to be changed, resulting in the modification being ineffective. This step is mainly to remind the user not to modify the property that does not exist.